src/main/scala/edu/berkeley/ce/rockslicing/Block.scala

```diff
@@ -62,19 +62,30 @@ object Block {
       DenseMatrix.eye[Double](3)
     }
   }
+}
+
+/**
+  * A rock block.
+  * @constructor Create a new rock block
+  * @param center Cartesian coordinates for the center of the rock block. The individual
+  * components can be accessed as 'centerX', 'centerY', and 'centerZ'.
+  * @param faces: The faces that define the boundaries of the rock block.
+  */
+case class Block(center: (Double,Double,Double), faces: Seq[Face]) {
+  val (centerX, centerY, centerZ) = center
+
+  // Computing a bounding sphere is expensive, so we only do it if necessary
+  lazy val ((sphereCenterX, sphereCenterY, sphereCenterZ), sphereRadius) = findBoundingSphere
+  // Computing the maximum radius of an inscribed sphere is also expensive
+  lazy val maxInscribableRadius = findMaxInscribableRadius
```

Comment:
This is pretty cool. I'm guessing it's okay to define `findBoundingSphere` and `findMaxInscribableRadius` after they've been called because of `lazy`?